drivers/dma/dma_stm32_v1.c

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright (c) 2019 Song Qiang <songqiang1304521@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @brief DMA low level driver implementation for F2/F4/F7 series SoCs.
+ */
+
+#include <drivers/dma.h>
+#include <soc.h>
+
+#define LOG_LEVEL CONFIG_DMA_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(dma_stm32_v1);
+
+/* DMA burst length */
+#define BURST_TRANS_LENGTH_1			0
+
+u32_t table_stream[] = {
+	LL_DMA_STREAM_0,
+	LL_DMA_STREAM_1,
+	LL_DMA_STREAM_2,
+	LL_DMA_STREAM_3,
+	LL_DMA_STREAM_4,
+	LL_DMA_STREAM_5,
+	LL_DMA_STREAM_6,
+	LL_DMA_STREAM_7,
+};
+
+static u32_t table_channel[] = {
+	LL_DMA_CHANNEL_0,
+	LL_DMA_CHANNEL_1,
+	LL_DMA_CHANNEL_2,
+	LL_DMA_CHANNEL_3,
+	LL_DMA_CHANNEL_4,
+	LL_DMA_CHANNEL_5,
+	LL_DMA_CHANNEL_6,
+	LL_DMA_CHANNEL_7,
+};
+
+void (*func_clear_ht[])(DMA_TypeDef *DMAx) = {
```

Comment:
+1 about these tables.
About the naming, what about `func_ll_` to make it clear in the calling code ?

Reply:
Sure

Reply:
I will also rename table_stream and table_channel to table_ll_stream and table_ll_channel since they are also table of LL defines.